core/api/src/services/openai/index.ts

```diff
@@ -0,0 +1,172 @@
+import OpenAI, { OpenAIError } from "openai"
+
+import { env } from "@/config/env"
+import { sleep } from "@/utils"
+import { UnknownDomainError } from "@/domain/shared"
+import { ChatSupportError, UnknownChatSupportError } from "@/domain/chat-support/errors"
+
+let openai: OpenAI
+
+if (env.OPENAI_API_KEY) {
+  openai = new OpenAI({
+    apiKey: env.OPENAI_API_KEY,
+  })
+}
+
+interface Assistant {
```

Comment:
this must be at domain layer and implemented in line 34, also please create arg types and follow naming conventions